docs/spec/v1beta2/imagerepositories.md

```diff
@@ -0,0 +1,723 @@
+# Image Repositories
+
+The `ImageRepository` API defined a repository to scan and store a specific set
+of tags in a database.
+
+## Example
+
+The following is an example of an ImageRepository. It scans the specified image
+repository and stores the scanned tags in an internal database.
+
+```yaml
+---
+apiVersion: image.toolkit.fluxcd.io/v1beta2
+kind: ImageRepository
+metadata:
+  name: podinfo
+  namespace: default
+spec:
+  image: stefanprodan/podinfo
+  interval: 1h
```

Comment:
Let's add `provider: generic` to this example and to the explanation below (linking to the provider section).